src/hooks/useYObserver.tsx

```diff
@@ -1,73 +1,109 @@
-import { useEffect, useState } from 'react'
+import * as yMapValueByPath from '@/lib/yMapValueByPath'
+import { useCollaboration } from '.'
+import { useCallback, useEffect, useState } from 'react'
 import * as Y from 'yjs'
 import { toYMap } from '../../src-srv/utils/transformations/lib/toYMap'
+import { type Block } from '@/protos/service'
 
-import { get, set } from '../lib/yMapValueByPath'
+export const useForceUpdate = (): () => void => {
+  const [, dispatch] = useState(Object.create(null))
 
-type YSetter = (value: unknown) => void
+  return useCallback(
+    () => dispatch(Object.create(null)),
+    [dispatch]
+  )
+}
+
+interface YObserved {
+  get: (key: string) => unknown | undefined
+  set: (value: string | Partial<Block> | undefined, key?: string) => void
+  state: Block | Block[]
+  loading: boolean
+}
+
+export function useYObserver(path: string): YObserved {
+  const [loading, setLoading] = useState(true)
+  const forceUpdate = useForceUpdate()
 
+  // Get Y.Doc from provider and extract planning Y.Map
+  const { provider, synced: isSynced } = useCollaboration()
+  const document = isSynced ? provider?.document : undefined
+  const yPlanning = document?.getMap('planning')
 
-/*
- * Create an observer for YJS Shared Type
- * Should always return a "workable" format, array, object or string
- */
-export function useYObserver<T>(y: Y.Map<unknown> | undefined, key: string): [T, YSetter]
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function useYObserver<T extends Record<string, any>>(y: Y.Map<unknown> | undefined, key?: string): [Record<string, T>, YSetter]
-export function useYObserver<T>(y: Y.Array<unknown> | undefined, key?: string): [T[], YSetter]
-export function useYObserver<T>(y: Y.Map<unknown> | Y.Array<unknown> | undefined, key?: string): [T | Record<string, T> | T[], YSetter] {
-  const [value, setValue] = useState<unknown | undefined>(get(y as Y.Map<unknown>, key))
+  // Get wanted Y.Map by path provided
+  const map = yMapValueByPath.get(yPlanning, path)
 
+  // Observe whole yPlanning to detect changes at top
```

Comment:
Are the two comments misleading? It's not just planning we're tracking?

Reply:
I would say it's accurate, it's just the planning/yPlanning we're observing. We're getting the planning ymap and attaching the observeDeep on that:
```
const yPlanning = document?.getMap('planning')
...
    yPlanning?.observeDeep((events) => {
```
Changes to `open-documents` and `content` shouldn't be observed afaik.

It's possible that the second observer (on the `map`) is superfluous. We should be able to catch all the changes on the observeDeep.

Reply:
Now i see, I'll fix so we also support the `article` Y.Map (and others)